mobile/src/utils/functions.ts

```diff
@@ -0,0 +1,247 @@
+  } else {
+    try {
+      const payload: { publicKey: string; otpCode?: string } = {
+        publicKey: pubKey,
+      };
+      if (!isNil(otpCode)) {
+        payload.otpCode = otpCode;
+      }
+      const result = await paymentsApi.post("/api/customers/devices", payload);
+      if (result.status === 200) {
+        if (result.data?.value?.otpSeed) {
+          await SecureStore.setItemAsync(
+            oid + "otpSeed",
+            result.data.value.otpSeed
+          );
+          await SecureStore.setItemAsync(oid + "deviceVerified", "true");
+          return true;
+        } else {
+          return result.data;
+        }
+      }
+    } catch (error) {
+      const axiosErrror = error as AxiosError;
+      if (axiosErrror.response?.status == 409) {
+        return { data: { error: "conflict" } };
+      } else {
+        return { data: { error: "error" } };
+      }
+    }
+  }
+};
+export const registerDevice = async (pubKey: string, oid: string) => {
+  const response = await verifyDevice(pubKey, oid, null);
+  if (response) {
+    if (response === true) {
+      return true;
+    } else {
+      if (response?.data?.error === "conflict") {
+        return "conflict";
+      } else if (response?.data?.error === "error") {
+        return "error";
+      } else {
+        return "error";
+      }
+    }
+  } else {
+    return false;
+  }
+};
+export const checkStoredKeys = async (oid: string) => {
+  if (isNil(oid)) {
+    return false;
+  }
+
+  const publicKey = await SecureStore.getItemAsync(oid + "_publicKey");
+  const privateKey = await SecureStore.getItemAsync(oid + "_privateKey");
+  const otpSeed = await SecureStore.getItemAsync(oid + "otpSeed");
+
+  if (isNil(publicKey) || isNil(privateKey)) {
+    const keys = await renewKeys(oid);
+    // verify device
+    const pubKey = keys.pubKey;
+    const deviceRegistered = await registerDevice(pubKey, oid);
+    return deviceRegistered;
+  }
+  if (isNil(otpSeed)) {
+    const deviceRegistered = await registerDevice(publicKey, oid);
+    return deviceRegistered;
+  }
+};
+
+export const getOid = async (): Promise<string | false> => {
+  const oid = await SecureStore.getItemAsync("oid");
+  if (isNil(oid)) {
+    return false;
+  }
+  return oid;
+};
+
+export const removeAllStoredData = async (whocallthisfunc: string) => {
+  // check if there is an oid
+  const oid = await SecureStore.getItemAsync("oid");
+  await SecureStore.deleteItemAsync(oid + "_publicKey");
+  await SecureStore.deleteItemAsync(oid + "_privateKey");
+  await SecureStore.deleteItemAsync(oid + "_deviceRegistered");
+  await SecureStore.deleteItemAsync("oid");
+  await SecureStore.deleteItemAsync("email");
+  await SecureStore.deleteItemAsync("phoneNumber");
+  await SecureStore.deleteItemAsync(oid + "RegistrationCompleted");
+  await SecureStore.deleteItemAsync(oid + "deviceVerified");
+  console.log("All stored data removed. request from: ", whocallthisfunc);
```

Comment:
Please remove all `console.log` statements.

Reply:
The function removed.